### ISSUE TITLE
Building chromaprint-1.4.2 fails

### ISSUE THREAD

**Issue by real-t0mg** [acoustid/chromaprint#39]
Building **chromaprint-1.4.2** fails with the following message:

```
-- Using swresample for audio conversion
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/SBo/chromaprint-1.4.2/build
Scanning dependencies of target chromaprint_objs
[  4%] Building CXX object src/CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o
In file included from /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.h:14:0,
                 from /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:4:
/tmp/SBo/chromaprint-1.4.2/src/utils.h: In function ‘void chromaprint::PrepareHammingWindow(RandomAccessIterator, RandomAccessIterator, double)’:
/tmp/SBo/chromaprint-1.4.2/src/utils.h:40:8: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/utils.h:40:13: error: ‘size’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/utils.h:42:61: error: ‘size’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/utils.h: In function ‘void chromaprint::ApplyWindow(InputIt, InputIt, WindowIt&, OutputIt&)’:
/tmp/SBo/chromaprint-1.4.2/src/utils.h:49:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/utils.h:49:7: error: ‘input’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/utils.h:50:9: error: ‘input’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp: In member function ‘void chromaprint::FFTLib::Load(const int16_t*, const int16_t*, const int16_t*, const int16_t*)’:
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:27:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:27:7: error: ‘window’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:28:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:28:7: error: ‘output’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:29:22: error: ‘window’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:29:30: error: ‘output’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp: In member function ‘void chromaprint::FFTLib::Compute(chromaprint::FFTFrame&)’:
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:35:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:35:7: error: ‘input’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:36:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:36:7: error: ‘output’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:37:2: error: ‘output’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:37:14: error: ‘input’ was not declared in this scope
make[2]: *** [src/CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/chromaprint_objs.dir/all] Error 2
make: *** [all] Error 2
root@stinkpad:/home/ftp/MIX/chromaprint# 
```

I'm building in my custom Slackware 13.37 x86_64 with GCC 4.5.2 against ffmpeg 3.0.7, FFTW 3.3.4 and cmake 3.5.2. I'm not a programmer but I can build my own packages, let me know what I may try to help debug this issue. Cheers, t0mg

**Comment by lalinsky** [comment]
Please run `make VERBOSE=1`. It seems to me that it's calling gcc without `-std=c++11` for some reason.

**Comment by real-t0mg** [comment]
Hi Lukas, 

I don't see any hint with verbose compile enabled. Any hint is appreciated.

```
root@stinkpad:/tmp/SBo/chromaprint-1.4.2# date 
Thu Feb 23 10:09:28 ART 2017
root@stinkpad:/tmp/SBo/chromaprint-1.4.2# cmake -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_TOOLS=ON -VERBOSE=1 && make 
-- The C compiler identification is GNU 4.5.2
-- The CXX compiler identification is GNU 4.5.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Failed
-- Looking for lrintf
-- Looking for lrintf - found
-- Looking for round
-- Looking for round - found
-- Looking for av_packet_unref
-- Looking for av_packet_unref - found
-- Looking for av_frame_alloc
-- Looking for av_frame_alloc - found
-- Looking for av_frame_free
-- Looking for av_frame_free - found
-- Found KissFFT: /tmp/SBo/chromaprint-1.4.2/vendor/kissfft  
-- Using avfft for FFT calculations
-- Using swresample for audio conversion
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/SBo/chromaprint-1.4.2
Scanning dependencies of target chromaprint_objs
[  4%] Building CXX object src/CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o
In file included from /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.h:14:0,
                 from /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:4:
/tmp/SBo/chromaprint-1.4.2/src/utils.h: In function ‘void chromaprint::PrepareHammingWindow(RandomAccessIterator, RandomAccessIterator, double)’:
/tmp/SBo/chromaprint-1.4.2/src/utils.h:40:8: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/utils.h:40:13: error: ‘size’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/utils.h:42:61: error: ‘size’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/utils.h: In function ‘void chromaprint::ApplyWindow(InputIt, InputIt, WindowIt&, OutputIt&)’:
/tmp/SBo/chromaprint-1.4.2/src/utils.h:49:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/utils.h:49:7: error: ‘input’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/utils.h:50:9: error: ‘input’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp: In member function ‘void chromaprint::FFTLib::Load(const int16_t*, const int16_t*, const int16_t*, const int16_t*)’:
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:27:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:27:7: error: ‘window’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:28:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:28:7: error: ‘output’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:29:22: error: ‘window’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:29:30: error: ‘output’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp: In member function ‘void chromaprint::FFTLib::Compute(chromaprint::FFTFrame&)’:
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:35:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:35:7: error: ‘input’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:36:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:36:7: error: ‘output’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:37:2: error: ‘output’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:37:14: error: ‘input’ was not declared in this scope
make[2]: *** [src/CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/chromaprint_objs.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by lalinsky** [comment]
Please run `make VERBOSE=1` instead of `make`.

**Comment by lalinsky** [comment]
Hm, but now I see `Performing Test COMPILER_SUPPORTS_CXX11 - Failed` so I'm not sure about the compiler. GCC 4.5.2 probably doesn't fully support C++11. Chromaprint 4.x needs a C++11 compiler.

**Comment by real-t0mg** [comment]
Understood. For completeness, let me paste the verbose make error prior to closing the issue. 

```
-- Found Threads: TRUE  
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Failed
-- Looking for lrintf
-- Looking for lrintf - found
-- Looking for round
-- Looking for round - found
-- Looking for av_packet_unref
-- Looking for av_packet_unref - found
-- Looking for av_frame_alloc
-- Looking for av_frame_alloc - found
-- Looking for av_frame_free
-- Looking for av_frame_free - found
-- Found KissFFT: /tmp/SBo/chromaprint-1.4.2/vendor/kissfft  
-- Using avfft for FFT calculations
-- Using swresample for audio conversion
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/SBo/chromaprint-1.4.2/build
/usr/bin/cmake -H/tmp/SBo/chromaprint-1.4.2 -B/tmp/SBo/chromaprint-1.4.2/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /tmp/SBo/chromaprint-1.4.2/build/CMakeFiles /tmp/SBo/chromaprint-1.4.2/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory `/tmp/SBo/chromaprint-1.4.2/build'
make -f src/CMakeFiles/chromaprint_objs.dir/build.make src/CMakeFiles/chromaprint_objs.dir/depend
make[2]: Entering directory `/tmp/SBo/chromaprint-1.4.2/build'
cd /tmp/SBo/chromaprint-1.4.2/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/SBo/chromaprint-1.4.2 /tmp/SBo/chromaprint-1.4.2/src /tmp/SBo/chromaprint-1.4.2/build /tmp/SBo/chromaprint-1.4.2/build/src /tmp/SBo/chromaprint-1.4.2/build/src/CMakeFiles/chromaprint_objs.dir/DependInfo.cmake --color=
Dependee "/tmp/SBo/chromaprint-1.4.2/build/src/CMakeFiles/chromaprint_objs.dir/DependInfo.cmake" is newer than depender "/tmp/SBo/chromaprint-1.4.2/build/src/CMakeFiles/chromaprint_objs.dir/depend.internal".
Dependee "/tmp/SBo/chromaprint-1.4.2/build/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/tmp/SBo/chromaprint-1.4.2/build/src/CMakeFiles/chromaprint_objs.dir/depend.internal".
Scanning dependencies of target chromaprint_objs
make[2]: Leaving directory `/tmp/SBo/chromaprint-1.4.2/build'
make -f src/CMakeFiles/chromaprint_objs.dir/build.make src/CMakeFiles/chromaprint_objs.dir/build
make[2]: Entering directory `/tmp/SBo/chromaprint-1.4.2/build'
[  4%] Building CXX object src/CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o
cd /tmp/SBo/chromaprint-1.4.2/build/src && /usr/bin/c++   -DHAVE_CONFIG_H -D_SCL_SECURE_NO_WARNINGS -D_USE_MATH_DEFINES -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/SBo/chromaprint-1.4.2/build -I/tmp/SBo/chromaprint-1.4.2/src  -O2 -march=native -fPIC -pipe -fomit-frame-pointer -Wall -O3 -DNDEBUG -fPIC   -o CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o -c /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp
In file included from /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.h:14:0,
                 from /tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:4:
/tmp/SBo/chromaprint-1.4.2/src/utils.h: In function ‘void chromaprint::PrepareHammingWindow(RandomAccessIterator, RandomAccessIterator, double)’:
/tmp/SBo/chromaprint-1.4.2/src/utils.h:40:8: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/utils.h:40:13: error: ‘size’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/utils.h:42:61: error: ‘size’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/utils.h: In function ‘void chromaprint::ApplyWindow(InputIt, InputIt, WindowIt&, OutputIt&)’:
/tmp/SBo/chromaprint-1.4.2/src/utils.h:49:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/utils.h:49:7: error: ‘input’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/utils.h:50:9: error: ‘input’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp: In member function ‘void chromaprint::FFTLib::Load(const int16_t*, const int16_t*, const int16_t*, const int16_t*)’:
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:27:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:27:7: error: ‘window’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:28:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:28:7: error: ‘output’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:29:22: error: ‘window’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:29:30: error: ‘output’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp: In member function ‘void chromaprint::FFTLib::Compute(chromaprint::FFTFrame&)’:
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:35:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:35:7: error: ‘input’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:36:2: warning: ‘auto’ will change meaning in C++0x; please remove it
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:36:7: error: ‘output’ does not name a type
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:37:2: error: ‘output’ was not declared in this scope
/tmp/SBo/chromaprint-1.4.2/src/fft_lib_avfft.cpp:37:14: error: ‘input’ was not declared in this scope
make[2]: *** [src/CMakeFiles/chromaprint_objs.dir/fft_lib_avfft.cpp.o] Error 1
make[2]: Leaving directory `/tmp/SBo/chromaprint-1.4.2/build'
make[1]: *** [src/CMakeFiles/chromaprint_objs.dir/all] Error 2
make[1]: Leaving directory `/tmp/SBo/chromaprint-1.4.2/build'
make: *** [all] Error 2
root@stinkpad:/home/ftp/MIX/chromaprint# 
```

**Comment by lalinsky** [comment]
You might it to work by editing CMakeLists.txt and changing `-std=c++11` to `-std=c++0x`, but no guarantees.
